// angular-toastify/fesm2015/angular-toastify.js
import * as i0 from '@angular/core';
import { EventEmitter, Injectable, Component, ViewChild, Input, Output, HostListener, ChangeDetectionStrategy, NgModule } from '@angular/core';
import { Subject } from 'rxjs';
import * as i1 from '@angular/common';
import { CommonModule } from '@angular/common';

let toastIdentitySequence = 0;
class Toast {
    constructor(message, type) {
        this.message = message;
        this.type = type;
        this.time = new Date().getTime();
        this.id = toastIdentitySequence++;
        this.$resetToast = new Subject();
    }
}

var ToastType;
(function (ToastType) {
    ToastType[ToastType["info"] = 0] = "info";
    ToastType[ToastType["success"] = 1] = "success";
    ToastType[ToastType["warning"] = 2] = "warning";
    ToastType[ToastType["error"] = 3] = "error";
    ToastType[ToastType["default"] = 4] = "default";
})(ToastType || (ToastType = {}));

class ToastService {
    constructor() {
        this.toastAddedEvent = new EventEmitter();
        this.dismissAllEvent = new EventEmitter();
    }
    dismissAll() {
        this.dismissAllEvent.emit();
    }
    info(message) {
        const toast = new Toast(message, ToastType.info);
        this.toastAddedEvent.emit(toast);
    }
    success(message) {
        const toast = new Toast(message, ToastType.success);
        this.toastAddedEvent.emit(toast);
    }
    warn(message) {
        const toast = new Toast(message, ToastType.warning);
        this.toastAddedEvent.emit(toast);
    }
    error(message) {
        const toast = new Toast(message, ToastType.error);
        this.toastAddedEvent.emit(toast);
    }
    default(message) {
        const toast = new Toast(message, ToastType.default);
        this.toastAddedEvent.emit(toast);
    }
}
/** @nocollapse */ ToastService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ToastService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
/** @nocollapse */ ToastService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ToastService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ToastService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return []; } });

class ToastifyToastComponent {
    constructor(_cd, _zone) {
        this._cd = _cd;
        this._zone = _zone;
        this.autoClose = 5000;
        this.hideProgressBar = false;
        this.pauseOnHover = true;
        this.pauseOnVisibilityChange = true;
        this.closeOnClick = true;
        this.dismissEvent = new EventEmitter();
        this.ToastType = ToastType;
        this.running = false;
    }
    ngOnInit() {
        this.autoCloseRemaining = this.autoClose;
        this.startTime = this.toast.time;
        this.toast.$resetToast.subscribe(() => this.resetToastTimer());
        // Do not start timer when toast is prompted while window is out of focus
        if (this.handleVisibilityChange && document.visibilityState === "visible") {
            this.startCloseTimer();
        }
        // Start progress bar animation
        this.triggerProgressBarAnimation();
    }
    triggerProgressBarAnimation() {
        // Cancel previous animlation to avoid leaks
        if (this._progressBarAnimation !== undefined) {
            cancelAnimationFrame(this._progressBarAnimation);
        }
        // Start animation
        const frame = () => {
            if (this.running) {
                const remainingTime = Math.max(0, this.expectedAutoDismissTime - new Date().getTime());
                const percentage = 100 - ((remainingTime / this.autoClose) * 100);
                this.progressBarCover.nativeElement.style.width = percentage + "%";
                if (percentage <= 0)
                    return;
            }
            this._progressBarAnimation = requestAnimationFrame(frame);
        };
        this._progressBarAnimation = requestAnimationFrame(frame);
    }
    ngOnDestroy() {
        var _a, _b;
        // Cancel animation
        if (this._progressBarAnimation) {
            cancelAnimationFrame(this._progressBarAnimation);
            this._progressBarAnimation = undefined;
        }
        // Clear auto close timeout
        this.clearTimerTimeout();
        // Complete all observables
        (_a = this.toast.$resetToast) === null || _a === void 0 ? void 0 : _a.complete();
        this.toast.$resetToast = null;
        (_b = this._$updateTimer) === null || _b === void 0 ? void 0 : _b.complete();
        this._$updateTimer = null;
    }
    startCloseTimer() {
        if (this.running || !this.autoClose) {
            return;
        }
        this.running = true;
        this.expectedAutoDismissTime =
            new Date().getTime() + this.autoCloseRemaining;
        this.autoDismissTimeout = this._zone.runOutsideAngular(() => setTimeout(() => {
            this._zone.run(() => {
                this.dismissEvent.emit();
                this._cd.markForCheck();
            });
        }, this.autoCloseRemaining));
    }
    pauseCloseTimer() {
        this.running = false;
        this.clearTimerTimeout();
        // Calculate the elapsed time, subtract remaining time
        this.pauseTime = new Date().getTime();
        const elapsed = this.pauseTime - this.startTime;
        this.autoCloseRemaining = this.autoClose - elapsed;
    }
    resetToastTimer() {
        this.clearTimerTimeout();
        this.running = false;
        this.startTime = new Date().getTime();
        this.autoCloseRemaining = this.autoClose;
        this.startCloseTimer();
    }
    clearTimerTimeout() {
        if (this.autoDismissTimeout !== undefined) {
            this.expectedAutoDismissTime = undefined;
            clearTimeout(this.autoDismissTimeout);
        }
    }
    handleDismissButtonAction() {
        if (this.closeOnClick) {
            return; // Let the other event handle the dismissal
        }
        this.clearTimerTimeout();
        this.dismissEvent.emit();
    }
    handleHostClick() {
        if (this.closeOnClick) {
            this.clearTimerTimeout();
            this.dismissEvent.emit();
        }
    }
    handleMouseEnter() {
        if (this.pauseOnHover) {
            this.pauseCloseTimer();
        }
    }
    handleMouseLeave() {
        if (this.pauseOnHover) {
            this.startCloseTimer();
            this.startTime = new Date().getTime() + (this.startTime - this.pauseTime);
        }
    }
    handleVisibilityChange() {
        if (!this.pauseOnVisibilityChange) {
            return;
        }
        if (document.visibilityState !== "visible") {
            this.pauseCloseTimer();
            this._cd.detectChanges();
        }
        else {
            this.startCloseTimer();
        }
    }
}
/** @nocollapse */ ToastifyToastComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ToastifyToastComponent, deps: [{ token: i0.ChangeDetectorRef }, { token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ ToastifyToastComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: ToastifyToastComponent, selector: "lib-toastify-toast", inputs: { autoClose: "autoClose", hideProgressBar: "hideProgressBar", pauseOnHover: "pauseOnHover", pauseOnVisibilityChange: "pauseOnVisibilityChange", closeOnClick: "closeOnClick", toast: "toast", iconLibrary: "iconLibrary" }, outputs: { dismissEvent: "dismissEvent" }, host: { listeners: { "click": "handleHostClick()", "mouseenter": "handleMouseEnter()", "mouseleave": "handleMouseLeave()", "document:visibilitychange": "handleVisibilityChange()" } }, viewQueries: [{ propertyName: "progressBar", first: true, predicate: ["progressBar"], descendants: true }, { propertyName: "progressBarCover", first: true, predicate: ["progressBarCover"], descendants: true }], ngImport: i0, template: "<div role=\"alert\" class=\"toast-body\">\n    <div class=\"icon-container\">\n        <span *ngIf=\"iconLibrary == 'material'\" [ngSwitch]=\"toast.type\">\n            <i class=\"material-icons\" *ngSwitchCase=\"ToastType.info\">info_outline</i>\n            <i class=\"material-icons\" *ngSwitchCase=\"ToastType.default\">info_outline</i>\n            <i class=\"material-icons\" *ngSwitchCase=\"ToastType.warning\">warning_outline</i>\n            <i class=\"material-icons\" *ngSwitchCase=\"ToastType.error\">error_outline</i>\n            <i class=\"material-icons\" *ngSwitchCase=\"ToastType.success\">done</i>\n        </span>\n        <span *ngIf=\"iconLibrary == 'font-awesome'\" [ngSwitch]=\"toast.type\">\n            <i *ngSwitchCase=\"ToastType.info\" class=\"fa fa-info\"></i>\n            <i *ngSwitchCase=\"ToastType.default\" class=\"fa fa-info\"></i>\n            <i *ngSwitchCase=\"ToastType.warning\" class=\"fa fa-exclamation-triangle\"></i>\n            <i *ngSwitchCase=\"ToastType.error\" class=\"fa fa-exclamation\"></i>\n            <i *ngSwitchCase=\"ToastType.success\" class=\"fa fa-check\"></i>\n        </span>\n    </div>\n    <div class=\"toast-container\">\n        <span>{{toast.message}}</span>\n    </div>\n</div>\n<button (click)=\"handleDismissButtonAction()\" class=\"close-button close-button--{{ToastType[toast.type]}}\" type=\"button\" aria-label=\"close\">\u2716</button>\n\n<div #progressBar *ngIf=\"!hideProgressBar\" class=\"progress-bar progress-bar&#45;&#45;{{ToastType[toast.type]}}\" style=\"opacity: 1;\"></div>\n<div #progressBarCover *ngIf=\"!hideProgressBar\" class=\"progress-bar-cover toast--{{ToastType[toast.type]}}\" [style.animation-duration]=\"this.autoClose + 'ms'\"  [style.animation-play-state]=\"running? 'running' : 'paused'\"></div>\n", styles: [".toast{position:relative;min-height:64px;box-sizing:border-box;margin-bottom:1rem;padding:8px;border-radius:1px;box-shadow:0 1px 10px 0 rgba(0,0,0,.1),0 2px 15px 0 rgba(0,0,0,.05);display:flex;justify-content:space-between;max-height:800px;overflow:hidden;font-family:sans-serif;cursor:pointer;direction:ltr}.toast--rtl{direction:rtl}.toast--default{background:#fff;color:#aaa}.toast--info{background:#3498db}.toast--success{background:#07bc0c}.toast--warning{background:#f1c40f}.toast--error{background:#e74c3c}.toast-body{margin:auto 0;flex:1}@media only screen and (max-width:480px){.toast{margin-bottom:0}}.close-button{color:#fff;font-weight:700;font-size:14px;background:transparent;outline:none;border:none;padding:0;cursor:pointer;opacity:.7;transition:.3s ease;align-self:flex-start}.close-button--default{color:#000;opacity:.3}.close-button:focus,.close-button:hover{opacity:1}.progress-bar-cover{right:0;z-index:100;direction:rtl}.progress-bar,.progress-bar-cover{position:absolute;bottom:0;width:100%;height:5px;transform-origin:left}.progress-bar{left:0;z-index:99;opacity:.7;background-color:hsla(0,0%,100%,.7)}.progress-bar--controlled{transition:transform .2s}.progress-bar--rtl{right:0;left:auto;transform-origin:right}.progress-bar--default{background:linear-gradient(90deg,#4cd964,#5ac8fa,#007aff,#34aadc,#5856d6,#ff2d55)}.icon-container,.toast-body{vertical-align:middle}.icon-container{display:inline-block;width:25px}.icon-container span{margin-right:.5rem;display:inline-block;text-align:center;width:20px}.icon-container span i{vertical-align:middle}.fa,.material-icons{font-size:18px}.toast-container{display:inline-block;width:calc(100% - 25px);vertical-align:middle}"], directives: [{ type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i1.NgSwitch, selector: "[ngSwitch]", inputs: ["ngSwitch"] }, { type: i1.NgSwitchCase, selector: "[ngSwitchCase]", inputs: ["ngSwitchCase"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ToastifyToastComponent, decorators: [{
            type: Component,
            args: [{
                    // tslint:disable-next-line:component-selector
                    selector: "lib-toastify-toast",
                    templateUrl: "./toastify-toast.component.html",
                    styleUrls: ["./toastify-toast.component.scss"],
                    // changeDetection: ChangeDetectionStrategy.OnPush
                }]
        }], ctorParameters: function () { return [{ type: i0.ChangeDetectorRef }, { type: i0.NgZone }]; }, propDecorators: { progressBar: [{
                type: ViewChild,
                args: ["progressBar"]
            }], progressBarCover: [{
                type: ViewChild,
                args: ["progressBarCover"]
            }], autoClose: [{
                type: Input
            }], hideProgressBar: [{
                type: Input
            }], pauseOnHover: [{
                type: Input
            }], pauseOnVisibilityChange: [{
                type: Input
            }], closeOnClick: [{
                type: Input
            }], toast: [{
                type: Input
            }], iconLibrary: [{
                type: Input
            }], dismissEvent: [{
                type: Output
            }], handleHostClick: [{
                type: HostListener,
                args: ["click"]
            }], handleMouseEnter: [{
                type: HostListener,
                args: ["mouseenter"]
            }], handleMouseLeave: [{
                type: HostListener,
                args: ["mouseleave"]
            }], handleVisibilityChange: [{
                type: HostListener,
                args: ["document:visibilitychange"]
            }] } });

class ToastifyToastContainerComponent {
    constructor(_toastService, _cd) {
        this._toastService = _toastService;
        this._cd = _cd;
        this.transitionDurations = 400;
        this.position = 'top-right';
        this.transition = 'bounce';
        this.autoClose = 5000;
        this.hideProgressBar = false;
        this.pauseOnHover = true;
        this.pauseOnVisibilityChange = true;
        this.closeOnClick = true;
        this.newestOnTop = false;
        this.preventDuplicates = false;
        this.iconLibrary = 'none';
        this.ToastType = ToastType;
        this.toasts = new Array();
        this.toastTransitionDict = {};
    }
    ngOnChanges() {
        this._cd.markForCheck();
    }
    dismiss(toast) {
        this.toastTransitionDict[toast.id] = TransitionState.exiting;
        setTimeout(() => {
            const index = this.toasts.indexOf(toast);
            this.toasts.splice(index, 1);
            this._cd.markForCheck();
        }, this.transitionDurations);
    }
    getClass(toast) {
        let base = `toast toast--${ToastType[toast.type]} `;
        const state = this.toastTransitionDict[toast.id];
        if (state === TransitionState.entering) {
            base += `${this.transition}-enter ${this.transition}-enter--${this.position}`;
        }
        else if (state === TransitionState.exiting) {
            base += `${this.transition}-exit ${this.transition}-exit--${this.position}`;
        }
        return base;
    }
    ngOnInit() {
        this._toastService.dismissAllEvent.subscribe(() => {
            this.toasts = new Array();
            this._cd.markForCheck();
        });
        this._toastService.toastAddedEvent.subscribe((toast) => this.handleToastAddedEvent(toast));
    }
    handleToastAddedEvent(toast) {
        // If "preventDuplicates" is enabled, toast is not added if one toast with same message exists
        // The only thing to do is to update the toast time to keep it visible
        if (this.preventDuplicates === true) {
            const sameToast = this.toasts.find((existingToast) => existingToast.message === toast.message);
            if (sameToast) {
                sameToast.$resetToast.next();
                this._cd.markForCheck();
                return;
            }
        }
        // Add new toast transition
        this.toastTransitionDict[toast.id] = TransitionState.entering;
        setTimeout(() => {
            this.toastTransitionDict[toast.id] = TransitionState.noTransition;
            this._cd.markForCheck();
        }, this.transitionDurations);
        // Add new toast
        if (this.newestOnTop) {
            this.toasts.unshift(toast);
        }
        else {
            this.toasts.push(toast);
        }
        this._cd.markForCheck();
    }
}
/** @nocollapse */ ToastifyToastContainerComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ToastifyToastContainerComponent, deps: [{ token: ToastService }, { token: i0.ChangeDetectorRef }], target: i0.ɵɵFactoryTarget.Component });
/** @nocollapse */ ToastifyToastContainerComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: ToastifyToastContainerComponent, selector: "lib-toastify-toast-container", inputs: { position: "position", transition: "transition", autoClose: "autoClose", hideProgressBar: "hideProgressBar", pauseOnHover: "pauseOnHover", pauseOnVisibilityChange: "pauseOnVisibilityChange", closeOnClick: "closeOnClick", newestOnTop: "newestOnTop", preventDuplicates: "preventDuplicates", iconLibrary: "iconLibrary" }, usesOnChanges: true, ngImport: i0, template: "<div class=\"toast-container toast-container--{{position}}\">\n    <lib-toastify-toast *ngFor=\"let toast of toasts\" [class]=\"getClass(toast)\" style=\"animation-fill-mode: both; animation-duration: 0.75s\"\n        [autoClose]=\"autoClose\" [toast]=\"toast\" (dismissEvent)=\"dismiss(toast)\" [hideProgressBar]=\"hideProgressBar\" [pauseOnHover]=\"pauseOnHover\"\n        [pauseOnVisibilityChange]=\"pauseOnVisibilityChange\" [closeOnClick]=\"closeOnClick\" [iconLibrary]=\"iconLibrary\">\n    </lib-toastify-toast>\n</div>", styles: [".toast-container{z-index:9999;-webkit-transform:translateZ(9999px);position:fixed;padding:4px;width:320px;box-sizing:border-box;color:#fff}.toast-container--top-left{top:1em;left:1em}.toast-container--top-center{top:1em;left:50%;margin-left:-160px}.toast-container--top-right{top:1em;right:1em}.toast-container--bottom-left{bottom:1em;left:1em}.toast-container--bottom-center{bottom:1em;left:50%;margin-left:-160px}.toast-container--bottom-right{bottom:1em;right:1em}@media only screen and (max-width:480px){.toast-container{width:100vw;padding:0;left:0;margin:0}.toast-container--top-center,.toast-container--top-left,.toast-container--top-right{top:0}.toast-container--bottom-center,.toast-container--bottom-left,.toast-container--bottom-right{bottom:0}.toast-container--rtl{right:0;left:auto}}.toast{position:relative;min-height:64px;box-sizing:border-box;margin-bottom:1rem;padding:8px;border-radius:1px;box-shadow:0 1px 10px 0 rgba(0,0,0,.1),0 2px 15px 0 rgba(0,0,0,.05);display:flex;justify-content:space-between;max-height:800px;overflow:hidden;font-family:sans-serif;cursor:pointer;direction:ltr}.toast--rtl{direction:rtl}.toast--default{background:#fff;color:#aaa}.toast--info{background:#3498db}.toast--success{background:#07bc0c}.toast--warning{background:#f1c40f}.toast--error{background:#e74c3c}.toast-body{margin:auto 0;flex:1}@media only screen and (max-width:480px){.toast{margin-bottom:0}}@keyframes bounceInRight{0%,60%,75%,90%,to{animation-timing-function:cubic-bezier(.215,.61,.355,1)}0%{opacity:0;transform:translate3d(3000px,0,0)}60%{opacity:1;transform:translate3d(-25px,0,0)}75%{transform:translate3d(10px,0,0)}90%{transform:translate3d(-5px,0,0)}to{transform:none}}@keyframes bounceOutRight{20%{opacity:1;transform:translate3d(-20px,0,0)}to{opacity:0;transform:translate3d(2000px,0,0)}}@keyframes bounceInLeft{0%,60%,75%,90%,to{animation-timing-function:cubic-bezier(.215,.61,.355,1)}0%{opacity:0;transform:translate3d(-3000px,0,0)}60%{opacity:1;transform:translate3d(25px,0,0)}75%{transform:translate3d(-10px,0,0)}90%{transform:translate3d(5px,0,0)}to{transform:none}}@keyframes bounceOutLeft{20%{opacity:1;transform:translate3d(20px,0,0)}to{opacity:0;transform:translate3d(-2000px,0,0)}}@keyframes bounceInUp{0%,60%,75%,90%,to{animation-timing-function:cubic-bezier(.215,.61,.355,1)}0%{opacity:0;transform:translate3d(0,3000px,0)}60%{opacity:1;transform:translate3d(0,-20px,0)}75%{transform:translate3d(0,10px,0)}90%{transform:translate3d(0,-5px,0)}to{transform:translateZ(0)}}@keyframes bounceOutUp{20%{transform:translate3d(0,-10px,0)}40%,45%{opacity:1;transform:translate3d(0,20px,0)}to{opacity:0;transform:translate3d(0,-2000px,0)}}@keyframes bounceInDown{0%,60%,75%,90%,to{animation-timing-function:cubic-bezier(.215,.61,.355,1)}0%{opacity:0;transform:translate3d(0,-3000px,0)}60%{opacity:1;transform:translate3d(0,25px,0)}75%{transform:translate3d(0,-10px,0)}90%{transform:translate3d(0,5px,0)}to{transform:none}}@keyframes bounceOutDown{20%{transform:translate3d(0,10px,0)}40%,45%{opacity:1;transform:translate3d(0,-20px,0)}to{opacity:0;transform:translate3d(0,2000px,0)}}.bounce-enter--bottom-left,.bounce-enter--top-left{animation-name:bounceInLeft}.bounce-enter--bottom-right,.bounce-enter--top-right{animation-name:bounceInRight}.bounce-enter--top-center{animation-name:bounceInDown}.bounce-enter--bottom-center{animation-name:bounceInUp}.bounce-exit--bottom-left,.bounce-exit--top-left{animation-name:bounceOutLeft}.bounce-exit--bottom-right,.bounce-exit--top-right{animation-name:bounceOutRight}.bounce-exit--top-center{animation-name:bounceOutUp}.bounce-exit--bottom-center{animation-name:bounceOutDown}@keyframes zoomIn{0%{opacity:0;transform:scale3d(.3,.3,.3)}50%{opacity:1}}@keyframes zoomOut{0%{opacity:1}50%{opacity:0;transform:scale3d(.3,.3,.3)}to{opacity:0}}.zoom-enter{animation-name:zoomIn}.zoom-exit{animation-name:zoomOut}@keyframes flipIn{0%{transform:perspective(400px) rotateX(90deg);animation-timing-function:ease-in;opacity:0}40%{transform:perspective(400px) rotateX(-20deg);animation-timing-function:ease-in}60%{transform:perspective(400px) rotateX(10deg);opacity:1}80%{transform:perspective(400px) rotateX(-5deg)}to{transform:perspective(400px)}}@keyframes flipOut{0%{transform:perspective(400px)}30%{transform:perspective(400px) rotateX(-20deg);opacity:1}to{transform:perspective(400px) rotateX(90deg);opacity:0}}.flip-enter{animation-name:flipIn}.flip-exit{animation-name:flipOut}@keyframes slideInRight{0%{transform:translate3d(110%,0,0);visibility:visible}to{transform:translateZ(0)}}@keyframes slideInLeft{0%{transform:translate3d(-110%,0,0);visibility:visible}to{transform:translateZ(0)}}@keyframes slideInUp{0%{transform:translate3d(0,110%,0);visibility:visible}to{transform:translateZ(0)}}@keyframes slideInDown{0%{transform:translate3d(0,-110%,0);visibility:visible}to{transform:translateZ(0)}}@keyframes slideOutRight{0%{transform:translateZ(0)}to{visibility:hidden;transform:translate3d(110%,0,0)}}@keyframes slideOutLeft{0%{transform:translateZ(0)}to{visibility:hidden;transform:translate3d(-110%,0,0)}}@keyframes slideOutDown{0%{transform:translateZ(0)}to{visibility:hidden;transform:translate3d(0,500px,0)}}@keyframes slideOutUp{0%{transform:translateZ(0)}to{visibility:hidden;transform:translate3d(0,-500px,0)}}.slide-enter--bottom-left,.slide-enter--top-left{animation-name:slideInLeft}.slide-enter--bottom-right,.slide-enter--top-right{animation-name:slideInRight}.slide-enter--top-center{animation-name:slideInDown}.slide-enter--bottom-center{animation-name:slideInUp}.slide-exit--bottom-left,.slide-exit--top-left{animation-name:slideOutLeft}.slide-exit--bottom-right,.slide-exit--top-right{animation-name:slideOutRight}.slide-exit--top-center{animation-name:slideOutUp}.slide-exit--bottom-center{animation-name:slideOutDown}"], components: [{ type: ToastifyToastComponent, selector: "lib-toastify-toast", inputs: ["autoClose", "hideProgressBar", "pauseOnHover", "pauseOnVisibilityChange", "closeOnClick", "toast", "iconLibrary"], outputs: ["dismissEvent"] }], directives: [{ type: i1.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ToastifyToastContainerComponent, decorators: [{
            type: Component,
            args: [{
                    // tslint:disable-next-line:component-selector
                    selector: 'lib-toastify-toast-container',
                    templateUrl: './toastify-toast-container.component.html',
                    styleUrls: ['./toastify-toast-container.component.scss'],
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: ToastService }, { type: i0.ChangeDetectorRef }]; }, propDecorators: { position: [{
                type: Input
            }], transition: [{
                type: Input
            }], autoClose: [{
                type: Input
            }], hideProgressBar: [{
                type: Input
            }], pauseOnHover: [{
                type: Input
            }], pauseOnVisibilityChange: [{
                type: Input
            }], closeOnClick: [{
                type: Input
            }], newestOnTop: [{
                type: Input
            }], preventDuplicates: [{
                type: Input
            }], iconLibrary: [{
                type: Input
            }] } });
var TransitionState;
(function (TransitionState) {
    TransitionState[TransitionState["entering"] = 0] = "entering";
    TransitionState[TransitionState["noTransition"] = 1] = "noTransition";
    TransitionState[TransitionState["exiting"] = 2] = "exiting";
})(TransitionState || (TransitionState = {}));

class AngularToastifyModule {
}
/** @nocollapse */ AngularToastifyModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AngularToastifyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ AngularToastifyModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AngularToastifyModule, declarations: [ToastifyToastComponent, ToastifyToastContainerComponent], imports: [CommonModule], exports: [ToastifyToastContainerComponent] });
/** @nocollapse */ AngularToastifyModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AngularToastifyModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AngularToastifyModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [ToastifyToastComponent, ToastifyToastContainerComponent],
                    imports: [CommonModule],
                    exports: [ToastifyToastContainerComponent]
                }]
        }] });

/*
 * Public API Surface of angular-toastify
 */

/**
 * Generated bundle index. Do not edit.
 */

export { AngularToastifyModule, ToastService, ToastifyToastContainerComponent };
