// angular-toastify/esm2015/lib/angular-toastify.module.js
import { NgModule } from '@angular/core';
import { ToastifyToastComponent } from './toastify-toast/toastify-toast.component';
import { ToastifyToastContainerComponent } from './toastify-toast-container/toastify-toast-container.component';
import { CommonModule } from '@angular/common';
import * as i0 from "@angular/core";
export class AngularToastifyModule {
}
/** @nocollapse */ AngularToastifyModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AngularToastifyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ AngularToastifyModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AngularToastifyModule, declarations: [ToastifyToastComponent, ToastifyToastContainerComponent], imports: [CommonModule], exports: [ToastifyToastContainerComponent] });
/** @nocollapse */ AngularToastifyModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AngularToastifyModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AngularToastifyModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [ToastifyToastComponent, ToastifyToastContainerComponent],
                    imports: [CommonModule],
                    exports: [ToastifyToastContainerComponent]
                }]
        }] });
